api/rest/restapi_test.go

```diff
@@ -570,6 +570,26 @@ func TestAPIPinEndpoint(t *testing.T) {
 	testBothEndpoints(t, tf)
 }
 
+func TestAPIPinEndpointWithPath(t *testing.T) {
+	rest := testAPI(t)
+	defer rest.Shutdown()
+
+	tf := func(t *testing.T, url urlF) {
+		var ci api.CidSerial
+
+		// test regular post
+		testPaths := test.Paths()
```

Comment:
I'd prefer to have testcases visisble here.

Reply:
We should test with some invalid paths and see if the errors we send to the user are meaningful. One thing that is going to come up is that we probably should 404 when paths are clearly wrong but right now we return 500.

Reply:
> I'd prefer to have testcases visible here

These testcases are used at multiple places, twice in method_tests and twice restapi_test, do you want to duplicate that much code?

Reply:
> I'd prefer to have testcases visisble here.

Actually, you were right about this. method_test and restapi_test would have different test cases.

Reply:
> One thing that is going to come up is that we probably should 404 when paths are clearly wrong but right now we return 500.

That would require us to have have some logic instead of relying on `ipfs resolve`. We have some logic at the moment at the moment, but that's only because it's necessary and ipfs go-path package isn't perfect. I think we can just use 500.